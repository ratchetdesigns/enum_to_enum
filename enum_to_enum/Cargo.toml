[package]
name = "enum_to_enum"
version = "0.1.0"
authors = ["Adam Berger <adam@ratchetdesigns.com>"]
license = "MIT"
description = "Derives possibly effectful conversions between enums"
include = ["README.md", "LICENSE-MIT", "src/**/*.rs"]
edition = "2018"

[dependencies]
enum_to_enum_derive = { version = "=0.1.0", path = "../enum_to_enum_derive"}
