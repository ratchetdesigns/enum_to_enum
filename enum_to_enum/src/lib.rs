#[allow(unused_imports)]
#[macro_use]
extern crate enum_to_enum_derive;

pub use enum_to_enum_derive::*;

#[derive(Debug, Clone)]
pub struct WithEffects<Value, Effect> {
    pub value: Value,
    pub effects: Vec<Effect>,
}
