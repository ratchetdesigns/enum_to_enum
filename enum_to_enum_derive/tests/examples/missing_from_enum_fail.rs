use enum_to_enum::FromEnum;

#[derive(Debug, Clone, PartialEq, Eq)]
enum Src {
    Case1(String),
}

#[derive(Debug, Clone, PartialEq, Eq, FromEnum)]
enum SimpleDest {
    Case1(String),
}

fn main() {}
