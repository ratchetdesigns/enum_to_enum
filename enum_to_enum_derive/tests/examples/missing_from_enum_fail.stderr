error: #[from_enum(Src)] must appear at least once to specify the source enum(s)
 --> $DIR/missing_from_enum_fail.rs:9:1
  |
9 | enum SimpleDest {
  | ^^^^
